src/pip/_internal/resolution/resolvelib/requirements.py

```diff
@@ -40,13 +41,23 @@ def is_satisfied_by(self, candidate: Candidate) -> bool:
 
 
 class SpecifierRequirement(Requirement):
-    def __init__(self, ireq: InstallRequirement) -> None:
+    def __init__(
+        self,
+        ireq: InstallRequirement,
+        *,
+        drop_extras: bool = False,
+    ) -> None:
+        """
+        :param drop_extras: Ignore any extras that are part of the install requirement,
+            making this a requirement on the base only.
+        """
         assert ireq.link is None, "This is a link, not a specifier"
-        self._ireq = ireq
-        self._extras = frozenset(ireq.extras)
+        self._drop_extras: bool = drop_extras
+        self._ireq = ireq if not drop_extras else install_req_drop_extras(ireq)
```

Comment:
Why is this `_drop_extras` attribute is needed? I imagine this can be done outsided of this class and not affect functionalities at all.

Reply:
You're right, it's an artifact of my previous implementation that I forgot to clean up. Thanks.